### PR TITLE
Fix Non-ready set for system.parts.

### PR DESCRIPTION
**Pull request by KochetovNicolai**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix `Not-ready Set` error while reading from `system.parts` (with `IN subquery`). Was introduced in #60510.

Fixes some cases from #52991

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit a7ff6d63c397e729919ab9e31d797b91b6ae4708 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/61666/a7ff6d63c397e729919ab9e31d797b91b6ae4708/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/Storages/System/StorageSystemDetachedParts.cpp [hunk 1]
```diff
@@ -320,16 +322,33 @@ class ReadFromSystemDetachedParts : public SourceStepWithFilter
...
 
-    const ActionsDAG::Node * predicate = nullptr;
+    ActionsDAGPtr filter;
     const size_t max_block_size;
     const size_t num_streams;
 };
 
 void ReadFromSystemDetachedParts::applyFilters(ActionDAGNodes added_filter_nodes)
 {
+    filter_actions_dag = ActionsDAG::buildFilterActionsDAG(added_filter_nodes.nodes);
```
Comment (antonio2368) [inline src/Storages/System/StorageSystemDetachedParts.cpp:332]: called twice?

File: src/Storages/System/StorageSystemDetachedParts.cpp [hunk 2]
```diff
@@ -320,16 +322,33 @@ class ReadFromSystemDetachedParts : public SourceStepWithFilter
...
+        block.insert(ColumnWithTypeAndName({}, std::make_shared<DataTypeString>(), "table"));
+        block.insert(ColumnWithTypeAndName({}, std::make_shared<DataTypeString>(), "engine"));
+        block.insert(ColumnWithTypeAndName({}, std::make_shared<DataTypeUInt8>(), "active"));
+        block.insert(ColumnWithTypeAndName({}, std::make_shared<DataTypeUUID>(), "uuid"));
+
+        filter = VirtualColumnUtils::splitFilterDagForAllowedInputs(predicate, &block);
+        if (filter)
+        {
+            auto empty_block = block.cloneWithColumns(block.cloneEmptyColumns());
+            VirtualColumnUtils::filterBlockWithDAG(filter, empty_block, context);
```
Comment (antonio2368) [inline src/Storages/System/StorageSystemDetachedParts.cpp:349]: can you explain what are we doing here?

Comment (KochetovNicolai) [reply src/Storages/System/StorageSystemDetachedParts.cpp]: It was needed for building sets.
I've changed this to call a separate function to make it more clear.

**Review by antonio2368 — APPROVED** [review]

**Pull request merged**